### PR TITLE
Fixed #33515 -- Prevented recreation of migration for ManyToManyField to lowercased swappable setting.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-33515

Thanks Chris Lee for the report.

Regression in 43289707809c814a70f0db38ca4f82f35f43dbfd.

Refs ticket-23916.

### CODE REVIEW

**Comment by felixxm** [comment]
@SentientClee Does it work for you?

File: django/db/models/fields/related.py [hunk 1]
```diff
@@ -1731,8 +1731,13 @@ def deconstruct(self):
         # Rel needs more work.
         if isinstance(self.remote_field.model, str):
             kwargs["to"] = self.remote_field.model
+            if "." in self.remote_field.model:
+                app_label, model_name = self.remote_field.model.split(".")
+                kwargs["to"] = "%s.%s" % (app_label, model_name.lower())
+            else:
+                kwargs["to"] = self.remote_field.model.lower()
```
Comment (carltongibson) [inline django/db/models/fields/related.py:1738]: ln1733 looks redundant. `kwargs["to"]` is reassigned on either branch here. 🤔

**Review by carltongibson — COMMENTED** [review]
If it were me I'd add a comment _why_ here.

File: django/db/models/fields/related.py (hunk 1 above)
Comment (felixxm) [reply django/db/models/fields/related.py:1738]: Yes, should be removed :facepalm:

**Comment by felixxm** [comment]
@carltongibson Thanks for checking :+1: I added a comment and removed redundant line.

**Comment by charettes** [comment]
I think you can actually ignore my comment about `through`, it seems `ManyToManyField` doesn't even have support for replacing it by a `SettingsReference` like it does with `to`.

**Comment by felixxm** [comment]
@charettes Thanks  :+1:

**Comment by SentientClee** [comment]
@felixxm I tested this code and it works as expected. Thank you for the quick review!

**Pull request merged**